Objects/dictobject.c

```diff
@@ -2504,6 +2559,14 @@ PyDict_Copy(PyObject *o)
             _PyObject_GC_TRACK(split_copy);
         return (PyObject *)split_copy;
     }
+
+    if (PyDict_CheckExact(o) && mp->ma_values == NULL &&
+            mp->ma_keys->dk_refcnt == 1 &&
+            mp->ma_used == mp->ma_keys->dk_nentries)
+    {
+        return clone_dict(o);
+    }
+
     copy = PyDict_New();
```

Comment:
Any reason not to at least use presized?

Reply:
This change isn't concerned with updating/touching the slow path.